### PR TITLE
README: explicit statement on single source build

### PR DESCRIPTION
**Pull request by Youw**
Make an explicit guarantee how HIDAPI can be used as a "single source" dependency.

### CODE REVIEW

**Comment by Youw** [comment]
I just realized there is an existing section "Integrating hidapi directly into your source tree".
I should probably merge it with the new "Embedding source code directly".

File: BUILD.md [hunk 1]
```diff
@@ -106,5 +107,26 @@ To build HIDAPI using MinGW or Cygwin using Autotools, use a general Autotools
...
+or you really think that embedding HIDAPI directly as a source into your project
+is the best available option for you - HIDAPI guarantees:
+- you need to include a _single source file_ into your project's build system,
+depending on your platform and the backend you want to use:
+    - [`windows\hid.c`](windows/hid.c);
+    - [`linux/hid.c`](windows/hid.c);
+    - [`libusb/hid.c`](windows/hid.c);
+    - [`libusb/hid.c`](windows/hid.c);
+    - [`mac/hid.c`](windows/hid.c);
+- add a [`hidapi`](hidapi) forlder to the include path when building `hid.c`;
```
Comment (netfab) [inline BUILD.md:128]: add a [`hidapi`](hidapi) **forlder** to the include

File: BUILD.md [hunk 2]
```diff
@@ -68,19 +68,31 @@ On Mac make sure you have XCode installed and its Command Line Tools.
...
+---
+If your project uses CMake as a build system, it is safe to add HIDAPI as a [subdirectory](BUILD.cmake.md#hidapi-as-a-subdirectory).
+
+---
+If _the only option_ that works for you is adding HIDAPI sources directly
+to your project's build system, then you need:
+- include a _single source file_ into your project's build system,
+depending on your platform and the backend you want to use:
+    - [`windows\hid.c`](windows/hid.c);
+    - [`linux/hid.c`](windows/hid.c);
```
Comment (JoergAtGithub) [inline BUILD.md:85]:
```suggestion
    - [`linux/hid.c`](linux/hid.c);
```
Same for the next lines

Comment (Youw) [reply BUILD.md]: omg, how did that happen

**Pull request merged**